src/legacy/ui/public/vis/editors/default/components/default_editor_agg_group.tsx

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect, useReducer } from 'react';
+import {
+  EuiTitle,
+  EuiDragDropContext,
+  EuiDroppable,
+  EuiDraggable,
+  EuiSpacer,
+  EuiPanel,
+} from '@elastic/eui';
+
+import { Vis } from '../../../';
+import { AggConfig } from '../../../agg_config';
+import { aggGroupNamesMap } from '../agg_groups';
+import { DefaultEditorAgg } from './default_editor_agg';
+import { DefaultEditorAggAdd } from './default_editor_agg_add';
+import { DefaultEditorAggCommonProps } from './default_editor_agg_common_props';
+import {
+  isInvalidAggsTouched,
+  isAggRemovable,
+  calcAggIsTooLow,
+} from './default_editor_agg_group_helper';
+import { aggGroupReducer, initAggsState, AGGS_ACTION_KEYS } from './default_editor_agg_group_state';
+import { Schema } from '../schemas';
+
+interface DefaultEditorAggGroupProps extends DefaultEditorAggCommonProps {
+  vis: Vis;
+  addSchema: (schems: Schema) => void;
+  reorderAggs: (group: AggConfig[]) => void;
+}
+
+function DefaultEditorAggGroup({
+  formIsTouched,
+  groupName,
+  lastParentPipelineAggTitle,
+  metricAggs,
+  state,
+  vis,
```

Comment:
The `vis` prop is only needed for searching available schemas `const schemas = vis.type.schemas[groupName];`
I suppose it's better to pass these schemas directly from the parent